src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -271,6 +271,8 @@ private void CancellationCallback()
                     finally
                     {
                         StaticWatcherRunLoopManager.UnscheduleFromRunLoop(_eventStream);
+                        _eventStream.Close();
+                        _eventStream = null;
```

Comment:
When can this be invoked?  Is it possible we could be nulling out _eventStream and have a race condition with someone else trying to dereference it?

Reply:
I did another check and I don't think so. we use _eventStream to start eventloop but it is not passed or used in callback with events.
CancellationCallback() has check for null and it calls FSEventStreamSto() first to stop loop and than we take steps to invalidate and release according to Apple's documentation.  
Only problem I could see is that the neither `_stopping` nor check for `_eventStream != null` is atomic.  Also `_disposed` is not set and check in atomic way in `FileSystemWatcher.cs`. 
I think there could be race if Dispose() is called concurrently from multiple threads. 
If you think it is worth it, I can assign `_eventStream` to local variable first and then check for null and call Close().